### DISCUSSION TITLE
Very difficult to review large PRs with GitHub

### DISCUSSION THREAD

**Discussion by chriskuech** [community/community#10830] [Pull Requests]
Problems:
* It is very hard to review large PRs, so I would like to add more controls to make this more manageable.
* It is very hard to track open conversations in large PRs.  The weird mix of content in the Conversations view is very confusing and makes all the content unhelpful.

Proposed features:
* A toggle for expanding all or contracting all files
* A file tree view like Azure DevOps shows.
* Adding an indicator at the top of a file showing that the file has an unresolved conversation in it.  It is currently very difficult to track unresolved conversations.
* Add file filter for unresolved conversations.
* Have a separate "Timeline" versus "Conversations" view.  Currently, the "Conversations" view is really a "Timeline" view and overloaded with non-conversation-related information, making the entire view overloaded and unusable.  It pushes useful info (PR checks) way out of view in favor of unuseful information (every commit ever in that PR).

**Comment by lfolco** [comment]
Any updates on this? This is making my job intolerable.

**Comment by 7thFox** [comment]
For anyone suffering with this I do have some JS scripts you can save as bookmarks. This is the only way GitHub is even mildly usable for non-trivial code reviews.

Check "Viewed" on all:
`javascript:boxes=document.getElementsByClassName("js-reviewed-checkbox");for(let e=0;e<boxes.length;e++){const t=boxes[e]; if (!t.checked) t.click();}`

Uncheck "Viewed" on all:
`javascript:boxes=document.getElementsByClassName("js-reviewed-checkbox");for(let e=0;e<boxes.length;e++){const t=boxes[e]; if (t.checked) t.click();}`

Load all diffs
`javascript:boxes=document.getElementsByClassName("load-diff-button"),counter=0;for(let e=0;e<boxes.length;e++){const t=boxes[e];t.click(),counter++};`

Load the next 100 diff (because why wouldn't loading 500 cause more performance problems?)
`javascript:boxes=document.getElementsByClassName("load-diff-button"),counter=0;for(let e=0;e<boxes.length && e<100;e++){const t=boxes[e];t.click(),counter++};`

Hope someone finds this helpful

**Comment by udlose** [comment]
Possibly fixed here: https://github.blog/changelog/2025-06-26-improved-pull-request-files-changed-experience-now-in-public-preview/

**Reply by cdaringe** [reply]
Still in preview :(

**Comment by tyler-reed-tsico** [comment]
Usually I refer to requesting the PR be broken up, but we often review the code changes made by linting tools in one swoop. Honestly, I don't mind the current PR format, but holy hell is the website performance slow.
